api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1547,8 +1547,24 @@ public ObjectStat statObject(String bucketName, String objectName, ServerSideEnc
       headers = sse.headers();
     }
 
+    checkBucketName(bucketName);
+    String errorMessage = "";
+    if ( objectName == null ) {
+      errorMessage += " Object name cannot be null.";
+    } else if ( objectName.equals("") ) {
+      errorMessage += " Object name cannot be empty.";
+    }
+    if ( !errorMessage.equals("") ) {
+      throw new InvalidArgumentException("Invalid Stat Object Argument(s)." + errorMessage);
+    }
     HttpResponse response = executeHead(bucketName, objectName, headers);
     ResponseHeader header = response.header();
+    if ( header.contentLength() == 0
+        && header.contentType() == null
+        && header.statusCode() == null ) {
```

Comment:
I don't think this check is useful.

According to S3 specification, `Content-Length` must be returned else its server bug. `statusCode` is always available (else you get an exception) and `contentType`can be treated as optional.